include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp

```diff
@@ -0,0 +1,108 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::affine_cell_proxy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/core/tuple_utility.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A wrapper for an affine score matrix cell.
+ * \implements seqan3::tuple_like
+ * \ingroup pairwise_alignment
+ *
+ * \details
+ *
+ * This wrapper provides a uniform access to the different elements of the cell within an affine score matrix. This includes
+ * the optimal score, the horizontal gap score and the vertical gap score.
+ */
+template <tuple_like tuple_t>
+class affine_cell_proxy : public tuple_t
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Defaulted.
+    affine_cell_proxy() noexcept(std::is_nothrow_default_constructible_v<tuple_t>)= default;
```

Comment:
I think the noexcept is added by the compiler by default if possible (which will include the tuple_t check I guess)?

Reply:
basically you are right. I just changed it yesterday after I was reading cppreference code where they explicitly document it like this. So never change to late stuff that is working :)